samples/interactive-tutorials/src/main/java/search/SearchWithBoostSpec.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START retail_search_product_with_boost_spec]
+
+/*
+ * Call Retail API to search for a products in a catalog, rerank the
+ * results boosting or burying the products that match defined condition.
+ */
+
+package search;
+
+import com.google.cloud.retail.v2.SearchRequest;
+import com.google.cloud.retail.v2.SearchRequest.BoostSpec;
+import com.google.cloud.retail.v2.SearchRequest.BoostSpec.ConditionBoostSpec;
+import com.google.cloud.retail.v2.SearchResponse;
+import com.google.cloud.retail.v2.SearchServiceClient;
+import java.io.IOException;
+import java.util.UUID;
+
+public class SearchWithBoostSpec {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectNumber = System.getenv("PROJECT_NUMBER");
+    String defaultCatalogName =
+        String.format("projects/%s/locations/global/catalogs/default_catalog", projectNumber);
+    String defaultSearchPlacementName = defaultCatalogName + "/placements/default_search";
+
+    search(defaultSearchPlacementName);
+  }
+
+  public static SearchResponse search(String defaultSearchPlacementName) throws IOException {
+    // TRY DIFFERENT CONDITIONS HERE:
```

Comment:
nit: it's preferable to link to any documentation on what these values are and what they can do

Reply:
We have the tutorials, published in google3, in that tutorials we lead user through the service features description and provide different values which can demonstrate the feature in a particular catalog data